src/main/java/org/apache/commons/io/IOUtils.java

```diff
@@ -790,16 +842,342 @@ public static boolean contentEqualsIgnoreEOL(final Reader input1, final Reader i
+                    }
+                case '\n':
+                    switch (charArray2[nowCheck2]) {
+                        case '\n':
+                            lastState1 = lastState2 = LastState.newLine;
+                            nowCheck1++;
+                            nowCheck2++;
+                            continue;
+                        case '\r':
+                            nowCheck2++;
+                            if (nowPos2 == nowCheck2) {
+                                if (nowCheck2 == CONTENT_EQUALS_CHAR_ARRAY_BUFFER_SIZE) {
+                                    nowPos2 = nowCheck2 = 0;
+                                }
+                                do {
+                                    nowRead2 = input2.read(charArray2, nowPos2,
+                                            CONTENT_EQUALS_CHAR_ARRAY_BUFFER_SIZE - nowPos2);
+                                } while (nowRead2 == 0);
+                                if (nowRead2 == -1) {
+                                    readEnd2 = true;
+                                } else {
+                                    nowPos2 += nowRead2;
+                                }
+                            }
+                            lastState1 = lastState2 = LastState.newLine;
+                            nowCheck1++;
+                            if (readEnd2) {
+                                continue;
+                            }
+                            if (charArray2[nowCheck2] == '\n') {
+                                nowCheck2++;
+                            }
+                            continue;
+                        default:
+                            if (lastState1 == LastState.r) {
+                                lastState1 = LastState.newLine;
+                                nowCheck1++;
+                                continue;
+                            } else {
+                                return false;
+                            }
+                    }
+                default:
+                    switch (charArray2[nowCheck2]) {
+                        case '\n':
+                            if (lastState2 == LastState.r) {
+                                lastState2 = LastState.newLine;
+                                nowCheck2++;
+                                continue;
+                            } else {
+                                return false;
+                            }
+                        case '\r':
+                            return false;
+                        default:
+                            if (charArray1[nowCheck1] != charArray2[nowCheck2]) {
+                                return false;
+                            }
+                            lastState1 = lastState2 = LastState.normal;
+                            nowCheck1++;
+                            nowCheck2++;
+                            continue;
+                    }
+            }
         }
```

Comment:
This is one giant complex method, wow. I can't see wanting to maintain it myself. I am not sure if exploding methods like this from simple, short, and easy to understand to complex, large, and hard to grok is what we want. To make this worse from a maintenance POV, there are zero comments in hundreds of lines of code; I can imagine someone seeing this as an opportunity to refactoring all the the repeated patterns into reusable methods.

Reply:
I agree with Gary on this one.  In some cases performance gains do not outweigh complexity of the code.

Reply:
@garydgregory @melloware 
> there are zero comments in hundreds of lines of code

comments just added now:)
please see the latest pr.

> In some cases performance gains do not outweigh complexity of the code.
Yes.
It is also out of my thought.
I thought it SHOULD be also a 500%+ faster like the contentEquals function, but it is actually only 92% faster in jmh.
So it might not worth that much inline.
I will try to split some sub-functions out later today, and hope that can reduce the size of this function.

Reply:
@garydgregory @melloware 
split done.
now it is shorter...
far shorter.